core/src/main/scala/sttp/client3/internal/ToRfc2616Converter.scala

```diff
@@ -0,0 +1,63 @@
+package sttp.client3.internal
+
+import sttp.client3._
+import sttp.model._
+
+class ToRfc2616Converter[R <: RequestT[Identity, _, _]] {
+  def apply(request: R): String = {
+    val extractMethod = request.method.method
+    val extractUri = request.uri
+    val result = s"$extractMethod $extractUri"
+    val headers = extractHeaders(request)
+    val resultWithHeaders = if (headers.isEmpty) result else result + s"\n$headers"
+    val body = extractBody(request)
+    if (body.isEmpty) resultWithHeaders else resultWithHeaders + s"\n\n$body"
+  }
+
+  private def extractBody(r: R): String = {
+    r.body match {
+      case StringBody(text, _, _) => s"$text"
+      case ByteArrayBody(_, _)    => "<PLACEHOLDER>"
+      case ByteBufferBody(_, _)   => "<PLACEHOLDER>"
+      case InputStreamBody(_, _)  => "<PLACEHOLDER>"
+      case StreamBody(_)          => "<PLACEHOLDER>"
+      case MultipartBody(parts)   => handleMultipartBody(parts) + "--<PLACEHOLDER>--"
+      case FileBody(file, _)      => s"<${file.name}"
+      case NoBody                 => ""
+    }
+  }
+
+  def handleMultipartBody(parts: Seq[Part[RequestBody[_]]]): String = {
+    parts
+      .map { p =>
+        p.body match {
+          case StringBody(s, _, _) =>
+            s"""--<PLACEHOLDER>
```

Comment:
shouldn't we create a proper boundary, instead of using a fixed one? I think `HttpURLConnectionBackend` contains an implementation

Reply:
In `toCurl` version boundaries are not required. Here I can create random boundary.

Reply:
Added :+1: